woocommerce/woocommerce-android
# Issue/10364 show current theme

Closes: #10364 


### Description
This PR adds these changes:
1. It adds loading the current theme when the picker is shown from the app settings.
2. It updates the UI depending on whether it's shown from store creation or app settings.
3. It adds unit tests.

The PR seems big, but it's due to the tests, which should be easy to review.

### Testing instructions
1. Use a WPCom site.
2. Open the app settings.
3. Click on Themes.
4. Confirm the current theme is shown as expected.

### Images/gif

| Store Creation | App Settings |
| --- | --- |
| ![Screenshot_20231212_105550](https://github.com/woocommerce/woocommerce-android/assets/1657201/74795dff-4706-4b47-842c-e90e15f16059) | ![Screenshot_20231212_110245](https://github.com/woocommerce/woocommerce-android/assets/1657201/adff2138-4318-47dd-9830-09d5eef83401) |
| ![Screenshot_20231212_105558](https://github.com/woocommerce/woocommerce-android/assets/1657201/5508fed3-2072-4bf8-bf59-1f0891f6ca39) | ![Screenshot_20231212_110250](https://github.com/woocommerce/woocommerce-android/assets/1657201/b72d03f7-87b8-49c5-b547-c3f83d5e98b3) |
